core/src/main/kotlin/entity/interaction/Interaction.kt

```diff
@@ -109,13 +109,11 @@ sealed class InteractionCommand {
             val firstLevelOptions = data.options.orEmpty()
             val rootPredicate = firstLevelOptions.isEmpty() || firstLevelOptions.any { it.value.value != null }
             val groupPredicate = firstLevelOptions.any { it.subCommands.orEmpty().isNotEmpty() }
-            val subCommandPredicate = firstLevelOptions.any { it.values.orEmpty().isNotEmpty() }
 
             return when {
                 rootPredicate -> RootCommand(data)
                 groupPredicate -> GroupCommand(data)
-                subCommandPredicate -> SubCommand(data)
-                else -> error("Undefined data structure")
+                else -> SubCommand(data) // if not root, or group, it's a sub-command
```

Comment:
While throwing an error isn't great, assuming that the structure is a sub-command because it didn't match the previous predicates is going to come back to haunt us when Discord decides to add another structure type.

Reply:
I'm aware, but I assumed if they add another structure type it won't have the same structure as the Sub-command it will fail.
The other way to do it was to do exhaustive checks on the data structures which are redundant if the structure will fail at the serialization level

Reply:
> I'm aware, but I assumed if they add another structure type it won't have the same structure as the Sub-command it will fail.

I'd rather it fail-fast with a descriptive error though.

> The other way to do it was to do exhaustive checks on the data structures which are redundant if the structure will fail at the serialization level

we don't make the distinction at the ser level right now though